src/util/ArrayUtils.ts

```diff
@@ -0,0 +1,10 @@
+export function toggleIncluded<T>(elt:T, included:T[]):T[] {
+    const index = included.indexOf(elt);
```

Comment:
Hmm, how about something like this instead?:

```suggestion
    let toggled = _.without(included, elt);
    
    if (toggled.length === included.length) {
        toggled = included.concat([elt]);
    }
    
    return toggled;
```